elastic/kibana
# [Metrics UI][7.5] Fix space issue with TSVB requests from Metrics UI

## Summary

This PR fixes an issue with TSVB requests originating from the Metrics UI. The Metrics UI tries to recreate the request to TSVB based on the incoming request. When the request object is instantiated the wrong Base Path is used due to how the base path is injected. This PR overrides `getBasePath()` on the request object to return the original base path. This will allow the space to be honored.

### Testing

1. Create a new space that only displays the Metrics UI
2. Create a role that has `read` access to `metricbeat-*` index pattern. Also add the previously created space to this role and limit the read access to `Metrics`
3. Create a new user with the role you just created.
4. Start indexing Metricbeat data for you local machine
5. Visit the Metrics UI
6. View Metrics for your host.
7. You should see data in the charts.

### Checklist

Delete any items that are not applicable to this PR.

- ~~Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
- ~~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
- ~~[Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
-~~This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~
- ~~This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server)~~
- ~~This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
